server/security_integration_test.go

```diff
@@ -0,0 +1,246 @@
+	require.Equal(s.T(), apiKeyResponse.Url, fmt.Sprintf("https://app.convoy.io/app-portal/%s?groupID=%s&appId=%s", apiKeyResponse.Key, s.DefaultGroup.UID, app.UID))
+	require.Equal(s.T(), apiKeyResponse.Type, "app_portal")
+	require.Equal(s.T(), apiKeyResponse.AppID, app.UID)
+}
+
+func (s *SecurityIntegrationTestSuite) Test_RevokeAPIKey() {
+	expectedStatusCode := http.StatusOK
+
+	// Just Before.
+	apiKey, _ := testdb.SeedAPIKey(s.DB, s.DefaultGroup, uuid.NewString(), "test", "api")
+
+	url := fmt.Sprintf("/api/v1/security/keys/%s/revoke", apiKey.UID)
+
+	req := createRequest(http.MethodPut, url, nil)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+
+	// Deep assert
+	a, err := s.DB.APIRepo().FindAPIKeyByID(context.Background(), apiKey.UID)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), datastore.DeletedDocumentStatus, a.DocumentStatus)
+	require.True(s.T(), a.DeletedAt > 0)
+}
+
+func (s *SecurityIntegrationTestSuite) Test_GetAPIKeyByID() {
+	expectedStatusCode := http.StatusOK
+
+	// Just Before.
+	apiKey, _ := testdb.SeedAPIKey(s.DB, s.DefaultGroup, uuid.NewString(), "test", "api")
+
+	url := fmt.Sprintf("/api/v1/security/keys/%s", apiKey.UID)
+	req := createRequest(http.MethodGet, url, nil)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+
+	// Deep Assert.
+	var apiKeyResponse models.APIKeyByIDResponse
+	parseResponse(s.T(), w.Result(), &apiKeyResponse)
+	require.Equal(s.T(), apiKeyResponse.UID, apiKey.UID)
+}
+
+func (s *SecurityIntegrationTestSuite) Test_GetAPIKeyByID_APIKey_not_found() {
+	expectedStatusCode := http.StatusBadRequest
+
+	url := fmt.Sprintf("/api/v1/security/keys/%s", uuid.NewString())
+	req := createRequest(http.MethodGet, url, nil)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+}
+
+func (s *SecurityIntegrationTestSuite) Test_UpdateAPIKey() {
+	expectedStatusCode := http.StatusOK
+
+	// Just Before.
+	apiKey, _ := testdb.SeedAPIKey(s.DB, s.DefaultGroup, uuid.NewString(), "test", "api")
+
+	bodyStr := `{"role":{"type":"api","groups":["%s"]}}`
+	body := serialize(bodyStr, s.DefaultGroup.UID)
+
+	url := fmt.Sprintf("/api/v1/security/keys/%s", apiKey.UID)
+	req := createRequest(http.MethodPut, url, body)
+	w := httptest.NewRecorder()
+
+	// Act.
+	s.Router.ServeHTTP(w, req)
+
+	// Assert.
+	require.Equal(s.T(), expectedStatusCode, w.Code)
+
+	// Deep Assert.
+	a, err := s.DB.APIRepo().FindAPIKeyByID(context.Background(), apiKey.UID)
+	require.NoError(s.T(), err)
+
+	var apiKeyResponse models.APIKeyByIDResponse
+	parseResponse(s.T(), w.Result(), &apiKeyResponse)
+	require.Equal(s.T(), a.Role, apiKeyResponse.Role)
+	require.Equal(s.T(), apiKeyResponse.UID, apiKey.UID)
+}
+
+func (s *SecurityIntegrationTestSuite) Test_UpdateAPIKey_APIKey_not_found() {
```

Comment:
@danvixent Let's stay consistent with the naming of `Test_UpdateAPIKey_APIKeyNotFound`

Reply:
fixed